src/cargo/ops/registry.rs

```diff
@@ -86,10 +90,14 @@ fn verify_dependencies(pkg: &Package, registry_src: &SourceId)
             }
         } else if dep.source_id() != registry_src {
             if dep.source_id().is_registry() {
-                bail!("crates cannot be published to crates.io with dependencies sourced from other\n\
-                       registries either publish `{}` on crates.io or pull it into this repository\n\
-                       and specify it with a path and version\n\
-                       (crate `{}` is pulled from {}", dep.name(), dep.name(), dep.source_id());
+                // Block requests to send to a registry if it is not an alternative
+                // registry
+                if !registry_src.is_alt_registry() {
+                    bail!("crates cannot be published to crates.io with dependencies sourced from other\n\
+                           registries either publish `{}` on crates.io or pull it into this repository\n\
+                           and specify it with a path and version\n\
+                           (crate `{}` is pulled from {}", dep.name(), dep.name(), dep.source_id());
```

Comment:
While you're at it I think there's a missing `)` in this message, mind throwing it in there?

Reply:
I will make that change.